### ISSUE TITLE
Headers and Footers

### ISSUE THREAD

**Issue by gregavola** [Telerik-Verified-Plugins/NativePageTransitions#26]
We currently have a tab bar that positioned to the bottle of the. When navigate between pages, this plugin takes a screenshot of the page with the tab bar, and slides it out of view, and the places it right back when it's move from page to page. Is there anyway to keep this tab-bar out of view to act more like a native app (the header moves, but the tabbar doesn't move).

You mentioned in the code: ```TODO this can also be used for not scrolling fixed headers and footers```, so I'm wondering what can be done to do this. Any suggestions?

Greg

**Comment by EddyVerbruggen** [comment]
Hi Greg,

At the moment the plugin doesn't support partial page slides, but that TODO you found is indeed an indication we're planning to support it in a future release. I think we'll support it by adding optional params to the `slide` function. Something like `fixedPixelsTop` and `fixedPixelsBottom`. You'd use the latter to fix the tab bar.

It's not really planned yet, but I'll bring this up internally to discuss priorities. Keeping this open so we won't forget.

Thanks

**Comment by gregavola** [comment]
Hey Eddy,

Thanks for the quick response. We're kinda of in a bind now, and looking to roll this out (it's the only thing we have noticed that is a bit of a UX issue). Is there anything that can be done to push this forward? I would think that any one that has a tabbar in their application would suffer this issue, or perhaps it's not a concern for people?

Greg

**Comment by EddyVerbruggen** [comment]
Hi Greg,

I guess you'll need this on iOS first (Apple review and all). I think [I just nailed it](https://dl.dropboxusercontent.com/u/10061342/GithubShares/NativePageTransitions/issue26-PartialSlide.mov). Is this what your app needs as well?

I have not yet investigated a similar implementation for Android, but I'm sure it's possible.

**Comment by gregavola** [comment]
Our focus now is just on iOS right now for this feature (eventually for android). This is perfect and exactly what I need :-)

Anything I can to help test/try it out?

Thanks again for your help!

Greg

**Comment by EddyVerbruggen** [comment]
Yes, please try v0.2.8, the docs for the fixed header and footer (which you can use simultaneously if needed) is at the [Verified Plugins Marketplace](http://plugins.telerik.com/plugin/native-page-transitions).

I'd love to see this in a real app :+1:

**Comment by gregavola** [comment]
Eddy,

This works perfectly for iPhone 6, but for iPhone 6+, it flickers when using the new code for the bottom bar (almost like it's not picking up the frame size correctly): https://www.dropbox.com/s/57eukuqcsv82rxs/2015-01-05%2000.19.55.mp4?dl=0

Any ideas?

Greg

**Comment by gregavola** [comment]
For anyone else that has this problem - I was able to temporary solve it for the "Slide" method by adding the piece of code: 

```
UIScreen* mainScreen = [UIScreen mainScreen];
  CGFloat mainScreenHeight = mainScreen.bounds.size.height;
    
  CGFloat retinaFactor = DISPLAY_SCALE;
    
  // check if the height is 736, which makes it a 6+
  if (mainScreenHeight == 736.0) {
    retinaFactor = 3.0f;
  }
```

I assume that  it would be nice to add it to the ```define``` sections - however I was unable to get it work.

```
#define IS_RETINA_DISPLAY() [[UIScreen mainScreen] respondsToSelector:@selector(scale)] && [[UIScreen mainScreen] scale] == 2.0f
#define DISPLAY_SCALE IS_RETINA_DISPLAY() ? 2.0f : 1.0f
````

**Comment by EddyVerbruggen** [comment]
Hi Greg,

Please test the 0.2.9 version. It adds full compatibility for the iPhone 6 Plus, including this header/footer feature.

I tested it on the i6+ before, but with an incorrect splashscreen size, so I was testing a scaled-up version which did not show this problem.

Thanks,
Eddy

**Comment by EddyVerbruggen** [comment]
Android support for fixed header/footer just landed in 0.2.13, please give it a spin.
